lib/publiccloud/provider.pm

```diff
@@ -517,6 +536,25 @@ sub vault_api {
     return $res->json;
 }
 
+=head2 vault_api
+
+Wrapper around C<<$self->vault_api()>> to get retry capability.
+=cut
+sub vault_api {
+    my ($self, $path, %args) = @_;
+    my $ret;
+    my $tries = 3;
+
+    while ($tries-- > 0) {
+        eval {
+            $ret = $self->__vault_api($path, %args);
+        };
+        return $ret unless ($@);
+        sleep 10;
+    }
+    die($@);
```

Comment:
should we record info stating something like "Maximum number of Vault request exceeded. Check Vault Server is up and running" ? Maybe for us it's clear, but it could give some hints. Otherwise, LGTM

Reply:
done